addons/agent_mirrorpeer_controller.go

```diff
@@ -156,6 +157,53 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			return ctrl.Result{}, fmt.Errorf("few failures occurred while labeling RBD StorageClasses: %v", errs)
 		}
 	}
+
+	if mirrorPeer.Spec.Type == multiclusterv1alpha1.Async {
+		if mirrorPeer.Status.Phase == multiclusterv1alpha1.ExchangedSecret {
+			logger.Info("Cleaning up stale onboarding token", "Token", string(mirrorPeer.GetUID()))
+			err = deleteStorageClusterPeerTokenSecret(ctx, r.HubClient, r.SpokeClusterName, string(mirrorPeer.GetUID()))
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, nil
+		}
+		if mirrorPeer.Status.Phase == multiclusterv1alpha1.ExchangingSecret {
+			var token corev1.Secret
+			err = r.HubClient.Get(ctx, types.NamespacedName{Namespace: r.SpokeClusterName, Name: string(mirrorPeer.GetUID())}, &token)
+			if err != nil && !errors.IsNotFound(err) {
+				return ctrl.Result{}, err
+			}
+			if err == nil {
+				// TODO: Replace it with exported type from ocs-operator
+				type OnboardingTicket struct {
+					ID                string `json:"id"`
+					ExpirationDate    int64  `json:"expirationDate,string"`
+					StorageQuotaInGiB uint   `json:"storageQuotaInGiB,omitempty"`
+				}
+				var ticketData OnboardingTicket
+				err = json.Unmarshal(token.Data["storagecluster-peer-token"], &ticketData)
```

Comment:
Convert `storagecluster-peer-token` to a constant. We are using it at many places